mesh/mesh.go

```diff
@@ -0,0 +1,112 @@
+package mesh
+
+import (
+	"crypto"
+	"errors"
+	"github.com/spacemeshos/go-spacemesh/log"
+	"sync"
+)
+
+type Mesh interface {
+	AddLayer(layer *Layer) error
+	GetLayer(i int) (*Layer, error)
+	GetBlock(id BlockID) (*Block, error)
+	LocalLayerCount() uint32
+	LatestKnownLayer() uint32
+	SetLatestKnownLayer(idx uint32)
+	Close()
+}
+
+type Peer crypto.PublicKey
+
+type LayersDB struct {
+	layerCount       uint32
```

Comment:
why uint32? don't we expect to have billions of layers? can be changed later also...